lib/remote/configobjectutility.cpp

```diff
@@ -35,8 +37,13 @@ String ConfigObjectUtility::GetObjectConfigPath(const Type::Ptr& type, const Str
 	/* This may throw an exception the caller above must handle. */
 	String prefix = GetConfigDir();
 
-	return prefix + "/conf.d/" + typeDir +
-		"/" + EscapeName(fullName) + ".conf";
+	auto old (prefix + "/conf.d/" + typeDir + "/" + EscapeName(fullName) + ".conf");
+
+	if (Utility::PathExists(old)) {
+		return std::move(old);
+	}
+
+	return prefix + "/conf.d/" + typeDir + "/" + fullName.SubStr(0, 10) + SHA1(fullName) + ".conf";
```

Comment:
10 seems a bit short to me. Looking at https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits, I think we can safely assume a limit of 255 bytes. We don't have to max out that limit, but something around 80 should be enough to represent the full name for most setups. Also some separator between name and hash would be nice.